cunumeric/array.py

```diff
@@ -2301,6 +2301,7 @@ def dot(self, rhs, out=None):
             self,
             rhs,
             out=out,
+            casting="no",
```

Comment:
I assume this is consistent with NumPy's behavior, right?

Reply:
Yes